### PR TITLE
Use output path from manifest.yaml

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?

PR updates `collector.generate` task to use output_path from manifest.yaml. This invoke task is only run locally. 

### Motivation
To simplify BYOC workflow for OTel Agent.

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

**Review by CelianR — APPROVED** [review]
LGTM, Iiuc this is not used by Windows so it should work (if this can be used by Windows please use `pathlib` for the paths)

File: tasks/collector.py [hunk 1]
```diff
@@ -80,8 +82,32 @@ def generate(ctx):
...
 
-    # Clean the files with main* in comp/otelcol/collector-contrib/impl
+    # Read the output path from the manifest file
     impl_path = "./comp/otelcol/collector-contrib/impl"
+    output_path = None
+    try:
+        with open(config_path) as file:
+            manifest = yaml.safe_load(file)
+            output_path = manifest["dist"]["output_path"]
+    except Exception as e:
```
Comment (chouetz) [inline tasks/collector.py:92]: You could probably use a more granular exception here (either `FileNotFoundError` or `yaml.scanner.ScannerError` for instance) to precise what kind of error you expect to happen, wdyt?

File: tasks/collector.py [hunk 2]
```diff
@@ -80,8 +82,32 @@ def generate(ctx):
...
+            code=1,
+        ) from e
+
+    if output_path != impl_path:
+        components_source = os.path.join(output_path, "components.go")
+        components_dest = os.path.join(impl_path, "components.go")
+        print(f"Copying {components_source} to {components_dest}")
+        try:
+            shutil.copy(components_source, components_dest)
+        except Exception as e:
```
Comment (chouetz) [inline tasks/collector.py:104]: same here

**Review by chouetz — APPROVED** [review]
minor comments only

**Review by songy23 — APPROVED** [review]

File: tasks/collector.py (hunk 1 above)
Comment (dineshg13) [reply tasks/collector.py:92]: We are using Exception in rest of the function. I will revisit the pattern later.

**Pull request merged**